bisection.go

```diff
@@ -7,11 +7,14 @@ package optimize
 import "math"
 
 // Bisection is a Linesearcher that uses a bisection to find a point that
-// satisfies the strong Wolfe conditions with the given gradient constant and
-// function constant of zero. If GradConst is zero, it will be set to a reasonable
-// value. Bisection will panic if GradConst is not between zero and one.
+// satisfies the strong Wolfe conditions with the given curvature factor and
+// sufficient decrease factor of zero.
```

Comment:
I think that saying "no function decrease" without "factor" could be misleading/incorrect because in fact there is no function _increase_. Finishing the sentence after "Wolfe conditions" should be enough clear, we talk about the factors because we want to stress that there is only the curvature factor. But even if we use the decrease factor of zero, there is no reason to prevent the users from setting it to a positive value. So unless you object, in another PR I'd like to add DecreaseFactor to Bisection and trim the sentence after "Wolfe conditions". I'll wait for your comment before merging this PR, @btracey